docs/architecture/blazor-for-web-forms-developers/state-management.md

```diff
@@ -1,22 +1,98 @@
 ---
 title: State management
 description: Learn about different approaches for managing state in ASP.NET Web Forms and Blazor.
-author: danroth27
-ms.author: daroth
-ms.date: 09/11/2019
+author: csharpfritz
+ms.author: jefritz
+ms.date: 15/04/2020
 ---
 # State management
 
 [!INCLUDE [book-preview](../../../includes/book-preview.md)]
 
-*This content is coming soon.*
+State management is a key concept of Web Forms applications, facilitated through View State, Session State, Application State, and Postback features. These stateful features of the framework helped to hide the state management required for an application and allowed application developers to focus on delivering their functionality.  With ASP.NET Core and Blazor, some of these features have been relocated and some have been removed altogether.  In this chapter, we will review how to maintain state and deliver the same functionality with the new features in Blazor.
 
-<!--
-- View state
-- Session state
-- Local storage
-- App state
--->
+## Request State Management with ViewState
+
+When discussing state management regarding a Web Forms application, many developers will instantly flash to the concept of ViewState. In Web Forms, ViewState managed the state of the content between HTTP requests by sending a large encoded block of text back and forth to the browser. The ViewState field could be overwhelmed with content from a page containing many elements, expanding as large as several megabytes in size.
+
+With Blazor Server, each requested component's state  is persisted in memory on the server between requests. It will only be disposed when the user navigates away from the component.  All in-scope field values of the component are available between interactions with the server.
```

Comment:
I actually didn't realize that was the case, and now I'm curious what it would mean to have a single serve with a million users on it. What happens to the size of the server executable?

Reply:
Then your server would have 1M active circuits with their corresponding state. The exact scale characters will depend on your app, but our baseline scale out tests show that Blazor Server itself can scale out to 10s of thousands of concurrent users. See https://docs.microsoft.com/aspnet/core/blazor/host-and-deploy/server for more details.